### PR TITLE
fix(1D): add 1D export on basic objects and helpers

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: include/geode/geometry/basic_objects/segment.h [hunk 1]
```diff
@@ -82,7 +81,7 @@ namespace geode
         OwnerSegment< dimension >& operator=(
             OwnerSegment< dimension >&& other );
     };
-    ALIAS_2D_AND_3D( OwnerSegment );
+    ALIAS_1D_AND_2D_AND_3D( OwnerSegment );
```
Comment (panquez) [inline include/geode/geometry/basic_objects/segment.h:84]: Is direction a method of Segment1D, what does it mean?

Comment (BotellaA) [reply include/geode/geometry/basic_objects/segment.h:84]: Not sure.. Probably the direction on the segment like +1 or - 1

File: src/geode/geometry/distance.cpp [hunk 2]
```diff
@@ -636,6 +636,15 @@ namespace geode
...
     }
 
+    template double opengeode_geometry_api point_point_distance(
+        const Point1D&, const Point1D& );
+    template std::tuple< double, Point1D > opengeode_geometry_api
+        point_segment_distance( const Point1D&, const Segment1D& );
+    template std::tuple< double, Point1D, Point1D > opengeode_geometry_api
+        segment_segment_distance( const Segment1D&, const Segment1D& );
+    template std::tuple< double, Point1D > opengeode_geometry_api
+        point_line_distance( const Point1D&, const InfiniteLine1D& );
```
Comment (panquez) [inline src/geode/geometry/distance.cpp:646]: i think it is always 0. useful?

File: src/geode/geometry/projection.cpp [hunk 3]
```diff
@@ -72,6 +72,11 @@ namespace geode
         return point - plane.normal() * distance;
     }
 
+    template Point1D opengeode_geometry_api point_segment_projection(
+        const Point1D&, const Segment1D& );
+    template Point1D opengeode_geometry_api point_line_projection(
```
Comment (panquez) [inline src/geode/geometry/projection.cpp:77]: almost same remark

**Review by panquez — APPROVED** [review]

**Comment by panquez** [comment]
Bindings todo when needed

**Pull request merged**